stdlib/public/core/KeyPath.swift

```diff
@@ -118,12 +121,86 @@ public class AnyKeyPath: Hashable, _AppendKeyPath {
       }
     }
   }
+    
+  /*
+  The following pertains to 32-bit architectures only.
+  We assume everything is a valid pointer to a potential
+  _kvcKeyPathStringPtr except for the first 4KB page which is reserved
+  for the nil pointer. Note that we have to distinguish between a valid
+  keypath offset of 0, and the nil pointer itself.
+  We use maximumOffsetOn32BitArchitecture + 1 for this case.
+    
+  The variable maximumOffsetOn32BitArchitecture is duplicated in the two
+  functions below since having it as a global would make accesses slower,
+  given getOffsetFromStorage() gets called on each KeyPath read. Further,
+  having it as an instance variable in AnyKeyPath would increase the size
+  of AnyKeyPath by 8 bytes.
+  TODO: Find a better method of refactoring this variable if possible.
+  */
+
+  func assignOffsetToStorage(offset: Int) {
+    let maximumOffsetOn32BitArchitecture = 4094
```

Comment:
Should this be 4096?

Reply:
The way I'm storing this is:
0 -> Actual nil pointer
1 -> Valid offset of 1
...
4094 -> Valid offset of 4094
4095 -> Valid offset of 0
4096 would be a pointer to the first value of the next page.

Reply:
I agree with the comment below that I could store it as "offset + 1", which would mean:
0 -> Actual nil pointer
1 -> Valid offset of 0
...
4094 -> Valid offset of 4093
4095 -> Valid offset of 4094
4096 -> pointer to the first value of the next page.